examples/python/trigger-pagerduty-incident/handler/handler.py

```diff
@@ -0,0 +1,179 @@
+import sys, json, os
+import urllib3
+import requests
+import json
+
+# Surpress SSL warnings
+if(os.getenv("insecure_ssl")):
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+#
+### Paths and Endpoints
+### More examples and details here - https://v2.developer.pagerduty.com/docs/send-an-event-events-api-v2
+#
+PAGERDUTY_API_PATH='https://events.pagerduty.com/v2/enqueue'
+PD_CONFIG='/var/openfaas/secrets/pdconfig'
+class FaaSResponse:
+    """
+    FaaSResponse is a helper class to construct a properly formatted message returned by this function.
+    By default, OpenFaaS will marshal this response message as JSON.
+    """    
+    def __init__(self, status, message):
+        """
+        Arguments:
+            status {str} -- the response status code
+            message {str} -- the response message
+        """        
+        self.status=status
+        self.message=message
+
+class Pagerduty:
+    """
+    Pagerduty is a Class used to trigger Events in PagerDuty.
+    """    
+
+    def __init__(self,conn):
+        """
+        Arguments:
+            conn {session} -- connection to PagerDuty REST API
+        """
+        try:
+             with open(PD_CONFIG, 'r') as pdconfigfile:
+                pdconfig = json.load(pdconfigfile)
+                self.routingkey=pdconfig['routing_key']
+                self.event_action=pdconfig['event_action']
+        except OSError as err:
+            res = FaaSResponse('400','Could not read PagerDuty configuration > OSError: {0}'.format(err))
+            print(json.dumps(vars(res)))
+            raise
```

Comment:
See comment below on how I think this could be simplified, avoiding nested try/catch and (re)raising.

Reply:
Updated the exception handling to be more accurate and avoided nested/complex flows